project/Build.scala

```diff
@@ -598,12 +608,8 @@ object Build {
 
     var extraClasspath = s"$scalaLib${File.pathSeparator}$dottyLib"
     if ((decompile || printTasty) && !args.contains("-classpath")) extraClasspath += s"${File.pathSeparator}."
-    if (args0.contains("-with-compiler")) {
-      if (!isDotty.value) {
-        throw new MessageOnlyException("-with-compiler can only be used with a bootstrapped compiler")
```

Comment:
We should still have this error if we use the compiler compiled with the reference compiler. With compiler only works properly with the bootstrapped compiler.

Reply:
If the ABI wasn't broken at the same time, I think it should work, so I'll put a warning message instead of an error.